django/core/paginator.py

```diff
@@ -96,7 +96,7 @@ def _get_page_range(self):
         Returns a 1-based range of pages for iterating through within
         a template for loop.
         """
-        return list(six.moves.range(1, self.num_pages + 1))
+        return iter(six.moves.range(1, self.num_pages + 1))
```

Comment:
Does this have the problem of https://code.djangoproject.com/ticket/23088#comment:6 – I think it is unexpected if you can now only iterate over `page_range` once. Could you use `return six.moves.range(1, self.num_pages + 1))` which, if I understand it correctly, returns the memory-efficient and iterable version in both python versions?